fbw-common/src/wasm/systems/systems/src/misc/mod.rs

```diff
@@ -0,0 +1,230 @@
+            cargo_boarding_pct: 0.0,
+            cargo_deboarding_pct: 0.0,
+        }
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        self.is_enabled
+    }
+
+    pub fn boarding_state(&self) -> GsxState {
+        self.boarding_state
+    }
+
+    pub fn deboarding_state(&self) -> GsxState {
+        self.deboarding_state
+    }
+
+    pub fn pax_boarding(&self) -> i32 {
+        self.pax_boarding
+    }
+
+    pub fn pax_deboarding(&self) -> i32 {
+        self.pax_deboarding
+    }
+
+    pub fn cargo_boarding_pct(&self) -> f64 {
+        self.cargo_boarding_pct
+    }
+
+    pub fn cargo_deboarding_pct(&self) -> f64 {
+        self.cargo_deboarding_pct
+    }
+
+    pub fn set_is_enabled(&mut self, is_enabled: bool) {
+        self.is_enabled = is_enabled;
+    }
+
+    pub fn set_boarding_state(&mut self, boarding_state: GsxState) {
+        self.boarding_state = boarding_state;
+    }
+
+    pub fn set_deboarding_state(&mut self, deboarding_state: GsxState) {
+        self.deboarding_state = deboarding_state;
+    }
+
+    pub fn set_pax_boarding(&mut self, pax_boarding: i32) {
+        self.pax_boarding = pax_boarding;
+    }
+
+    pub fn set_pax_deboarding(&mut self, pax_deboarding: i32) {
+        self.pax_deboarding = pax_deboarding;
+    }
+
+    pub fn set_cargo_boarding_pct(&mut self, cargo_boarding_pct: f64) {
+        self.cargo_boarding_pct = cargo_boarding_pct;
+    }
+
+    pub fn set_cargo_deboarding_pct(&mut self, cargo_deboarding_pct: f64) {
+        self.cargo_deboarding_pct = cargo_deboarding_pct;
+    }
+}
+impl SimulationElement for Gsx {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.is_enabled = reader.read(&self.is_enabled_id);
+        self.pax_boarding = reader.read(&self.pax_boarding_id);
+        self.pax_deboarding = reader.read(&self.pax_deboarding_id);
+        self.cargo_boarding_pct = reader.read(&self.cargo_boarding_pct_id);
+        self.cargo_deboarding_pct = reader.read(&self.cargo_deboarding_pct_id);
+        self.boarding_state = reader.read(&self.boarding_state_id);
+        self.deboarding_state = reader.read(&self.deboarding_state_id);
+    }
+}
+
+pub struct BoardingSounds {
+    pax_board_id: VariableIdentifier,
+    pax_deboard_id: VariableIdentifier,
+    pax_complete_id: VariableIdentifier,
+    pax_ambience_id: VariableIdentifier,
+    pax_boarding: bool,
+    pax_deboarding: bool,
+    pax_complete: bool,
+    pax_ambience: bool,
+}
+impl BoardingSounds {
+    pub fn new(
+        pax_board_id: VariableIdentifier,
+        pax_deboard_id: VariableIdentifier,
+        pax_complete_id: VariableIdentifier,
+        pax_ambience_id: VariableIdentifier,
+    ) -> Self {
+        BoardingSounds {
+            pax_board_id,
+            pax_deboard_id,
+            pax_complete_id,
+            pax_ambience_id,
+            pax_boarding: false,
+            pax_deboarding: false,
+            pax_complete: false,
+            pax_ambience: false,
+        }
+    }
```

Comment:
Will do but I realise that Boris might have a different idea for how sounds for the A380/A320 may work, so this might be moved out of the shared component in the future.

Reply:
Well then we can move it again. But still think it's better to do this for now.